rotkehlchen/chain/ethereum/modules/lido_csm/balances.py

```diff
@@ -0,0 +1,117 @@
+import logging
+from collections import defaultdict
+from typing import TYPE_CHECKING
+
+from rotkehlchen.accounting.structures.balance import Balance, BalanceSheet
+from rotkehlchen.assets.utils import asset_normalized_value
+from rotkehlchen.chain.ethereum.interfaces.balances import BalancesSheetType, ProtocolWithBalance
+from rotkehlchen.chain.ethereum.modules.lido_csm.constants import (
+    ACCOUNTING_ABI,
+    CPT_LIDO_CSM,
+    LIDO_CSM_ACCOUNTING_CONTRACT,
+    STETH_ABI,
+)
+from rotkehlchen.chain.ethereum.modules.lido_csm.metrics import LidoCsmMetricsFetcher
+from rotkehlchen.chain.evm.contracts import EvmContract
+from rotkehlchen.constants.assets import A_STETH
+from rotkehlchen.constants.misc import ZERO
+from rotkehlchen.db.lido_csm import DBLidoCsm
+from rotkehlchen.errors.misc import RemoteError
+from rotkehlchen.fval import FVal
+from rotkehlchen.inquirer import Inquirer
+from rotkehlchen.logging import RotkehlchenLogsAdapter
+
+if TYPE_CHECKING:
+    from rotkehlchen.chain.ethereum.decoding.decoder import EthereumTransactionDecoder
+    from rotkehlchen.chain.ethereum.node_inquirer import EthereumInquirer
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+class LidoCsmBalances(ProtocolWithBalance):
+    """Queries Lido CSM accounting contract for node operator bond balances."""
+
+    def __init__(
+            self,
+            evm_inquirer: 'EthereumInquirer',
+            tx_decoder: 'EthereumTransactionDecoder',
+    ) -> None:
+        super().__init__(
+            evm_inquirer=evm_inquirer,
+            tx_decoder=tx_decoder,
+            counterparty=CPT_LIDO_CSM,
+            deposit_event_types=set(),
+        )
+        self._node_operator_db = DBLidoCsm(self.event_db.db)
+        self._steth_token = A_STETH.resolve_to_evm_token()
+        self._accounting_contract = EvmContract(
+            address=LIDO_CSM_ACCOUNTING_CONTRACT,
+            abi=ACCOUNTING_ABI,
+            deployed_block=0,
+        )
+        self._steth_contract = EvmContract(
+            address=self._steth_token.evm_address,
+            abi=STETH_ABI,
+            deployed_block=0,
+        )
+        # Reuse the metrics fetcher to compute pending rewards in stETH
+        self._metrics_fetcher = LidoCsmMetricsFetcher(evm_inquirer=evm_inquirer)
+
+    def _get_bond_shares(self, node_operator_id: int) -> int:
```

Comment:
can you add a docstring to this and other function in this file. Short one liner of what they do is fine. But also like we do in other parts in the code please write `May raise: ....`  followed by a list of Exceptions this function may raise. The idea we do this for is to know what exceptions to handle/check for handling all around the clal stack when we call a function.